hangman.py

```diff
@@ -1,213 +1,331 @@
-             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
-             that has already been revealed.
-
-    '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
-
-
-
-def hangman_with_hints(secret_word):
-    '''
-    secret_word: string, the secret word to guess.
-    
-    Starts up an interactive game of Hangman.
-    
-    * At the start of the game, let the user know how many 
-      letters the secret_word contains and how many guesses s/he starts with.
-      
-    * The user should start with 6 guesses
-    
-    * Before each round, you should display to the user how many guesses
-      s/he has left and the letters that the user has not yet guessed.
-    
-    * Ask the user to supply one guess per round. Make sure to check that the user guesses a letter
-      
-    * The user should receive feedback immediately after each guess 
-      about whether their guess appears in the computer's word.
-
-    * After each guess, you should display to the user the 
-      partially guessed word so far.
-      
-    * If the guess is the symbol *, print out all words in wordlist that
-      matches the current guessed word. 
-    
-    Follows the other limitations detailed in the problem write-up.
-    '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
-
-
-
-# When you've completed your hangman_with_hint function, comment the two similar
-# lines above that were used to run the hangman function, and then uncomment
-# these two lines and run this file to test!
-# Hint: You might want to pick your own secret_word while you're testing.
-
-
-if __name__ == "__main__":
-    # pass
-
-    # To test part 2, comment out the pass line above and
-    # uncomment the following two lines.
-    
-    secret_word = choose_word(wordlist)
-    hangman(secret_word)
-
-###############
-    
-    # To test part 3 re-comment out the above lines and 
-    # uncomment the following two lines. 
-    
-    #secret_word = choose_word(wordlist)
-    #hangman_with_hints(secret_word)
+# Problem Set 2, hangman.py
+# Name: Alexei Baida
+# Collaborators:
+# Time spent: 12+ hours
+
+import random
+import string
+
+WORDLIST_FILENAME = "words.txt"
+GUESSES_INITIAL = 6
+WARNINGS_INITIAL = 3
+VOWELS = ['a', 'e', 'i', 'o', 'u']
+HINT_SIGN = '*'
+UNDERSCORE = '_'
+
+
+class TextStyle:
+    """Class for styling text with different colour/font"""
+    DEFAULT = '\033[0m'
+    YELLOW = '\033[33m'
+    BOLD = '\033[1m'
+    RED = '\033[31m'
+    CHANGE_BACK = '\033[7m'
+
+    @classmethod
+    def yellow(cls, text):
+        return TextStyle.YELLOW + text + TextStyle.DEFAULT
+
+    @classmethod
+    def bold(cls, text):
+        return TextStyle.BOLD + text + TextStyle.DEFAULT
+
+    @classmethod
+    def red(cls, text):
+        return TextStyle.RED + text + TextStyle.DEFAULT
+
+    @classmethod
+    def change_background_font(cls, text):
+        return TextStyle.CHANGE_BACK + text + TextStyle.DEFAULT

```

Comment:
It reads a bit hard for me, when you use `CHANGE_BACK` and `DEFAULT` in one line, it's obvious the difference)

Reply:
Changed name of CHANGE_BACK to SWAP_COLOUR